src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,198 @@
+
+        Stopwatch requestStopwatch = Stopwatch.createStarted();
+        File csvFile = null;
+        File tmpDir = fileHelper.createTempDir();
+        File zipFile = null;
+
+        try {
+            // get caller's user using Bridge API getParticipantByIdForApp
+            StudyParticipant participant = bridgeHelper.getParticipant(appId, userId, false);
+            String orgMembership = participant.getOrgMembership();
+
+            // call Bridge API searchAccountSummariesForApp(appId, caller's Org)
+            int offsetBy = 0;
+            List<AccountSummary> accountSummaries = bridgeHelper.getAccountSummariesForApp(appId, orgMembership, offsetBy);
+
+            csvFile = fileHelper.newFile(tmpDir, getDownloadFilenamePrefix() + ".csv");
+            try (CSVWriter csvFileWriter = new CSVWriter(fileHelper.getWriter(csvFile))) { //TODO is this nested try catch block an anti pattern? :/
+                while (!accountSummaries.isEmpty()) {
+                    for (AccountSummary accountSummary : accountSummaries) {
+                        String[] row = getAccountSummaryArray(accountSummary);
+                        csvFileWriter.writeNext(row);
+                    }
+                    // get next set of account summaries
+                    offsetBy += PAGE_SIZE;
+                    accountSummaries = bridgeHelper.getAccountSummariesForApp(appId, orgMembership, offsetBy);
+
+                    // avoid burning out Bridge Server
+                    Thread.sleep(THREAD_SLEEP_INTERVAL);
+                }
+            } catch (IOException ex) {
+                throw new WorkerException("Error creating file " + csvFile + ": " + ex.getMessage(), ex);
+            }
+
+            // zip up and email the csv file
+            AppInfo appInfo = dynamoHelper.getApp(appId);
+            AccountInfo accountInfo = bridgeHelper.getAccountInfo(appId, userId);
+            String zipFileName = "userdata-" + UUID.randomUUID().toString() + ".zip";
+            zipFile = fileHelper.newFile(tmpDir, zipFileName);
+            zipFiles(csvFile, zipFile);
+            sesHelper.sendAttachmentToAccount(appInfo, accountInfo, zipFile.getAbsolutePath());
+
+        } catch (BridgeSDKException ex) {
+            int status = ex.getStatusCode();
+            if (status >= 400 && status < 500) {
+                throw new PollSqsWorkerBadRequestException(ex);
+            } else {
+                throw new RuntimeException(ex);
+            }
+        } finally {
+            fileHelper.deleteFile(csvFile);
+            fileHelper.deleteFile(zipFile);
+            fileHelper.deleteDir(tmpDir);
+            LOG.info("request took " + requestStopwatch.elapsed(TimeUnit.SECONDS) +
+                    " seconds for userId =" + userId + ", app=" + appId + ", password=" + password);
+        }
+    }
+
+    protected String getDownloadFilenamePrefix() {
+        return "account_summaries";
+    }
```

Comment:
Is this modeled after something? Do we need to make a method to return a string? Or can we make it a static final variable?